### PR TITLE
Deduplicate compile log statement warnings

### PR DESCRIPTION
**Pull request by matthew-mcallister**

### CODE REVIEW

**Comment by matthew-mcallister** [comment]
Ah, good thinking. I didn't even realize I'd made either of those oversights. I'm fine with making these changes, and I'll assume the modification to the test framework is small enough to just roll into this branch. Also I'll rebase after I'm done.

**Comment by matthew-mcallister** [comment]
OK, addressed this, and this time I checked the test results aren't obvious false positives.

Aside: I had the idea to add clang-style error formatting options so that the errors could be displayed one per line, and I started on a branch for that, but this particular case was easy to handle since the filename is the same each time. Perhaps that idea can be revisited another day.

File: src/parser.cpp [hunk 1]
```diff
@@ -2739,6 +2739,7 @@ static AstNode *ast_parse_async_prefix(ParseContext *pc) {
 
     AstNode *res = ast_create_node(pc, NodeTypeFnCallExpr, async);
     res->data.fn_call_expr.is_async = true;
+    res->data.fn_call_expr.seen = false;
```
Comment (andrewrk) [inline src/parser.cpp]: This is harmless, but you can rely on `ast_create_node zero` filling everything.

File: src/all_types.hpp [hunk 2]
```diff
@@ -648,6 +648,7 @@ struct AstNodeFnCallExpr {
     ZigList<AstNode *> params;
     bool is_builtin;
     bool is_async;
+    bool seen; // used by @compileLog
```
Comment (andrewrk) [inline src/all_types.hpp]: Let's just merge this as is, but one idea would be to rename this `seen_by_compile_log` and then the comment would be unnecessary.

**Comment by andrewrk** [comment]
Thanks!

File: src/all_types.hpp (hunk 2 above)
Comment (matthew-mcallister) [reply src/all_types.hpp]: Ah, I would have pushed again for that, but thanks anyways.

**Pull request merged**